src/classes/RD2_ValidationService.cls

```diff
@@ -285,7 +285,7 @@ public with sharing class RD2_ValidationService {
             } catch (Exception e) {
             }
 
-            Boolean isValid = dayOfMonth > 0 && dayOfMonth < 29;
+            Boolean isValid = dayOfMonth > 0 && dayOfMonth < 32;
```

Comment:
This should be `< 31`

Reply:
@npsp-reedestockton , see how PR #5286 is handling validation service and other similar common changes, so you can just copy over common changes required for the of both PRs. Of course, feel free to add new tests if required.